src/vivarium/framework/randomness/stream.py

```diff
@@ -305,12 +321,12 @@ def choice(
 
     def sample_from_distribution(
         self,
-        index: pd.Index,
-        distribution: Optional[stats.rv_continuous] = None,
-        ppf: Optional[Callable[[pd.Series, dict[str, Any]], pd.Series]] = None,
+        index: pd.Index[int],
+        distribution: stats.rv_continuous | None = None,
+        ppf: Callable[[pd.Series[Any], dict[str, Any]], pd.Series[Any]] | None = None,
```

Comment:
I'm pretty sure a `ppf` needs to have floats as input. So you should be able to type this as 
```suggestion
        ppf: Callable[[pd.Series[float], dict[str, Any]], pd.Series[Any]] | None = None,
```